test_suite/tests/ui/expected-string/integer.rs

```diff
@@ -2,8 +2,12 @@ use serde_derive::Serialize;
 
 #[derive(Serialize)]
 struct S {
-    #[serde(rename = 100)]
-    integer: (),
+    #[serde(rename = 100i128)]
```

Comment:
I think it might be nice to squash the test patch into this one, since it's a bit odd to have negative tests but no positive tests.